### PR TITLE
Regenerate `signedTxTestData`

### PR DESCRIPTION
**Pull request by Anviking**
- [x] Regenerate them in `BabbageEra` such that they can work with the `RecentEra` constraint.
- [x] Regenerate a whole lot more of them using the integration tests

### Comments

```
Cardano.Wallet.Shelley.Transaction
  estimateSignedTxSize
    equals the binary size of signed txs
      0.cbor [✔] (107ms)
      1.cbor [✔] (92ms)
      2.cbor [✔] (49ms)
      3.cbor [✔] (65ms)
      4.cbor [✔] (43ms)
      5.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      6.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      7.cbor [✔] (73ms)
      8.cbor [✔] (107ms)
      9.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      10.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      11.cbor [✔] (43ms)
      12.cbor [✔] (21ms)
      13.cbor [✔] (41ms)
      14.cbor [✔] (24ms)
      15.cbor [✔] (20ms)
      16.cbor [✔] (20ms)
      17.cbor [✔] (21ms)
      18.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      19.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      20.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      21.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      22.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      23.cbor [✔] (42ms)
      24.cbor [✔] (35ms)
      25.cbor [‐] (6ms)
        # PENDING: Test setup does not work for txs with scripts
      26.cbor [✔] (57ms)
      27.cbor [✔] (35ms)
      28.cbor [✔] (28ms)
      29.cbor [‐] (1ms)
        # PENDING: Test setup does not work for txs with scripts
      30.cbor [‐] (7ms)
        # PENDING: Test setup does not work for txs with scripts
      31.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      32.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      33.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      34.cbor [✔] (57ms)
      35.cbor [✔] (20ms)
      36.cbor [✔] (34ms)
      37.cbor [✔] (21ms)
      38.cbor [✔] (22ms)
      39.cbor [✔] (22ms)
      40.cbor [✔] (31ms)
      41.cbor [‐]
        # PENDING: Test setup does not work for txs with scripts
      42.cbor [‐] (23ms)
        # PENDING: Test setup does not work for txs with scripts
      43.cbor [✔] (38ms)
      44.cbor [✔] (21ms)
      45.cbor [✔] (20ms)
      46.cbor [✔] (21ms)
      47.cbor [✔] (45ms)
      48.cbor [✔] (25ms)
      49.cbor [✔] (30ms)
      50.cbor [✔] (20ms)
      51.cbor [✔] (30ms)
      52.cbor [✔] (22ms)
      53.cbor [✔] (25ms)
      54.cbor [✔] (23ms)
      55.cbor [✔] (33ms)
      56.cbor [✔] (22ms)
      57.cbor [✔] (22ms)
      58.cbor [✔] (37ms)
      59.cbor [✔] (21ms)
      60.cbor [✔] (32ms)
      61.cbor [✔] (21ms)
      62.cbor [✔] (20ms)
      63.cbor [✔] (35ms)
      64.cbor [✔] (20ms)
      65.cbor [✔] (35ms)
      66.cbor [✔] (20ms)
      67.cbor [✔] (21ms)
      68.cbor [✔] (43ms)
      69.cbor [✔] (23ms)
      70.cbor [✔] (43ms)
      71.cbor [✔] (42ms)
      72.cbor [✔] (43ms)
      73.cbor [✔] (48ms)
      74.cbor [✔] (24ms)
      75.cbor [✔] (29ms)
      76.cbor [✔] (25ms)
      77.cbor [✔] (49ms)
      78.cbor [✔] (50ms)
      79.cbor [✔] (20ms)
      80.cbor [✔] (22ms)
      81.cbor [✔] (41ms)
      82.cbor [✔] (28ms)
      83.cbor [✔] (53ms)
      84.cbor [✔] (28ms)
      85.cbor [✔] (37ms)
      86.cbor [✔] (46ms)
      87.cbor [✔] (27ms)
      88.cbor [✔] (22ms)
      89.cbor [✔] (21ms)
      90.cbor [✔] (22ms)
      91.cbor [✔] (27ms)
      92.cbor [✔] (20ms)
      93.cbor [✔] (27ms)
      94.cbor [✔] (20ms)
      95.cbor [✔] (42ms)
      96.cbor [✔] (49ms)
      97.cbor [✔] (41ms)
      98.cbor [✔] (49ms)
      99.cbor [✔] (45ms)
      100.cbor [✔] (20ms)
      101.cbor [✔] (51ms)
      102.cbor [✔] (29ms)
      103.cbor [✔] (25ms)
      104.cbor [✔] (71ms)
      105.cbor [✔] (73ms)
      106.cbor [✔] (22ms)
      107.cbor [✔] (22ms)
      108.cbor [✔] (56ms)
      109.cbor [✔] (71ms)
      110.cbor [✔] (20ms)
      111.cbor [✔] (22ms)
      112.cbor [✔] (25ms)
      113.cbor [✔] (20ms)
      114.cbor [✔] (25ms)
      115.cbor [✔] (27ms)
      116.cbor [‐] (16ms)
        # PENDING: Test setup does not work for txs with scripts
      117.cbor [‐] (4ms)
        # PENDING: Test setup does not work for txs with scripts
      118.cbor [‐] (14ms)
        # PENDING: Test setup does not work for txs with scripts
      119.cbor [✔] (40ms)
      120.cbor [✔] (63ms)
      121.cbor [✔] (27ms)
      122.cbor [✔] (51ms)
      123.cbor [✔] (22ms)
      124.cbor [✔] (23ms)
      125.cbor [✔] (26ms)
      126.cbor [✔] (44ms)
      127.cbor [✔] (56ms)
      128.cbor [✔] (42ms)
      129.cbor [✔] (47ms)
      130.cbor [✔] (21ms)
      131.cbor [✔] (25ms)
      132.cbor [✔] (22ms)
      133.cbor [✔] (36ms)
      134.cbor [✔] (21ms)
      135.cbor [‐]
        # PENDING: Test setup does not work for txs with bootstrap wits
      136.cbor [‐] (21ms)
        # PENDING: Test setup does not work for txs with bootstrap wits
      137.cbor [‐]
        # PENDING: Test setup does not work for txs with bootstrap wits
      138.cbor [‐] (6ms)
        # PENDING: Test setup does not work for txs with bootstrap wits
      139.cbor [✔] (23ms)
      140.cbor [✔] (21ms)
      141.cbor [✔] (24ms)
      142.cbor [✔] (21ms)
      143.cbor [✔] (43ms)
      144.cbor [✔] (52ms)
      145.cbor [✔] (24ms)
      146.cbor [✔] (24ms)
      147.cbor [✔] (31ms)
      148.cbor [✔] (40ms)
      149.cbor [✔] (31ms)
      150.cbor [✔] (21ms)
      151.cbor [✔] (23ms)
      152.cbor [✔] (34ms)

Finished in 0.7552 seconds, used 2.4894 seconds of CPU time
153 examples, 0 failures, 24 pending
```

### Issue Number

Targets #3873

### CODE REVIEW

File: lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs [hunk 1]
```diff
@@ -4641,28 +4645,70 @@ prop_updateTx
...
+                    (Alonzo.witsAlonzoTxL . Alonzo.scriptAlonzoWitsL)
+                RecentEraConway -> Map.null $ ledgerTx ^.
+                    (Alonzo.witsAlonzoTxL . Alonzo.scriptAlonzoWitsL)
+            noBootWits = case WriteTx.recentEra @era of
+                RecentEraAlonzo -> Set.null $ ledgerTx ^.
+                    (Alonzo.witsAlonzoTxL . Alonzo.bootAddrAlonzoWitsL)
+                RecentEraBabbage -> Set.null $ ledgerTx ^.
+                    (Alonzo.witsAlonzoTxL . Alonzo.bootAddrAlonzoWitsL)
+                RecentEraConway -> Set.null $ ledgerTx ^.
+                    (Alonzo.witsAlonzoTxL . Alonzo.bootAddrAlonzoWitsL)
```
Comment (Anviking) [inline lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs:4683]: I would suggest to go with:

```haskell
-- | Convenient constraints. May break as we move to new eras.
type RecentEraLedgerConstraints era =
    (Core.Era era, Core.Script era ~ AlonzoScript era)

class
    ( Cardano.IsShelleyBasedEra era
    , Typeable era
    , RecentEraLedgerConstraints (ShelleyLedgerEra era)
    ) => IsRecentEra era where
    recentEra :: RecentEra era
```
but that's for its own PR.

**Comment by paweljakubas** [comment]
can you highlight the procedure of regeneration? Is there a comment telling how it is done?

**Comment by Anviking** [comment]
> can you highlight the procedure of regeneration? Is there a comment telling how it is done?

There's a comment in this genData.patch patch:

https://github.com/input-output-hk/cardano-wallet/blob/0bf049d4ac1781aadf6a001e94029845db0f4033/lib/wallet/test/data/signedTxs/genData.patch

**Comment by paweljakubas** [comment]
what about writing this down or referencing the module in a docs? Somewhere where things like 
```
CREATE_MISSING_GOLDEN=TRUE cabal run cardano-wallet:test:unit -- --match "produces the same JSON"
```
or similar are placed?

**Comment by paweljakubas** [comment]
cases with script are pending now. They will need test rewrite to accommodate them, correct? Planning this in this PR?

**Comment by Anviking** [comment]
> Planning this in this PR?

I wasn't originally (this is already a strict improvement), but maybe... 🤔 The problem is we'd need a real (or real enough) `UTxO` set for looking up all inputs, but this _should_ be possible to retrieve in a similar hacky way from the integration tests... E.g. inserting some tmp code for `balanceTx` to write _all_ UTxO sets into one big union on-disk...

**Review by paweljakubas — APPROVED** [review]
modulo hlint lgtm

File: lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs [hunk 2]
```diff
@@ -4888,61 +4943,22 @@ modifyBabbageTxBody
...
-    , "6fbfc0a1884fb54636554d14945b799125cf7ce38d477f5158405835ff78c6fc5e"
-    , "4466a179ca659fa85c99b8a3fba083f3f3f42ba360d479c64ef169914b52ade49b"
-    , "19a7208fd63a6e67a19c406b4826608fdc5307025506c307825820010000000000"
-    , "00000000000000000000000000000000000000000000000000005840e8e769ecd0"
-    , "f3c538f0a5a574a1c881775f086d6f4c845b81be9b78955728bffa7efa54297c6a"
-    , "5d73337bd6280205b1759c13f79d4c93f29871fc51b78aeba80ef6"
-    ]
+-- | Goldens can be regenrated by running the integration tests with
+-- lib/wallet/test/data/signedTxs/genData.patch applied.
+signedTxGoldens :: IO [(FilePath, ByteString)]
```
Comment (Anviking) [inline lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs:4948]: Perhaps golden is the wrong name here? This is simply used as test data, not really to check that nothing's changed.

**Comment by Anviking** [comment]
> what about writing this down or referencing the module in a docs? Somewhere where things like

Added a comment about how to regenerate https://github.com/input-output-hk/cardano-wallet/pull/3881/commits/192e896d4f09ce55ca8ab069e7f89d858e75273f + "golden" might not have been the right wording https://github.com/input-output-hk/cardano-wallet/pull/3881#discussion_r1175640767

File: lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs (hunk 1 above)
Comment (Anviking) [reply lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs:4683]: #3882

**Review by jonathanknowles — APPROVED** [review]
Hi @Anviking 

This LGTM. I agree with your thoughts about the term "golden", since we usually use "golden" to describe an (input, output) value pair for some function `f`, rather than just an input value per se.

Perhaps something like "testData" is better after all.

**Comment by jonathanknowles** [comment]
@Anviking I've actually merged all commits from this PR into #3873.

However, GitHub doesn't seem to have noticed. 😄

**Comment by Anviking** [comment]
Manually merged, now to master.

Fixing the pending script and bootstrap cases will be for another time.

**Pull request closed without merging**